diode-server/reconciler/repository.go

```diff
@@ -0,0 +1,17 @@
+package reconciler
+
+import (
+	"context"
+
+	"github.com/netboxlabs/diode/diode-server/gen/diode/v1/reconcilerpb"
+	"github.com/netboxlabs/diode/diode-server/reconciler/changeset"
+)
+
+// Repository is an interface for interacting with ingestion logs and change sets.
+type Repository interface {
+	CreateIngestionLog(ctx context.Context, ingestionLog *reconcilerpb.IngestionLog, sourceMetadata []byte) (*int32, error)
+	UpdateIngestionLogStateWithError(ctx context.Context, id int32, state reconcilerpb.State, ingestionError *reconcilerpb.IngestionError) error
+	RetrieveIngestionLogs(ctx context.Context, filter *reconcilerpb.RetrieveIngestionLogsRequest, limit int32, offset int32) ([]*reconcilerpb.IngestionLog, error)
+	CountIngestionLogsPerState(ctx context.Context) (map[reconcilerpb.State]int32, error)
+	CreateChangeSet(ctx context.Context, changeSet changeset.ChangeSet, ingestionLogID int32) (*int32, error)
+}
```

Comment:
discussed a bit offline, would like to see this continue to evolve, possibly bury dbids inside here and/or think about the structures that go in and out separately from the grpc interface and how to address transactionality.